src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,243 @@
+            this.boxVisible(false);
+          }
+        });
+
+        this.registerInteraction(this._dragInteraction);
+      }
+
+      protected _setup() {
+        super._setup();
+
+        var createLine = () => this._box.append("line").style({
+                                 "opacity": 0,
+                                 "stroke": "pink"
+                               });
+        this._boxEdgeT = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeB = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeL = createLine().classed("drag-edge-lr", true);
+        this._boxEdgeR = createLine().classed("drag-edge-lr", true);
+
+        var createCorner = () => this._box.append("circle")
+                                     .style({
+                                       "opacity": 0,
+                                       "fill": "pink"
+                                     });
+        this._boxCornerTL = createCorner().classed("drag-corner-tl", true);
+        this._boxCornerTR = createCorner().classed("drag-corner-tr", true);
+        this._boxCornerBL = createCorner().classed("drag-corner-bl", true);
+        this._boxCornerBR = createCorner().classed("drag-corner-br", true);
+
+        this.detectionRadius(this._detectionRadius);
+      }
+
+      private _getEdges(p: Point) {
+        var edges = {
+          top: false,
+          bottom: false,
+          left: false,
+          right: false
+        };
+
+        var bounds = this.bounds();
+        var t = bounds.topLeft.y;
+        var b = bounds.bottomRight.y;
+        var l = bounds.topLeft.x;
+        var r = bounds.bottomRight.x;
+        var rad = this._detectionRadius;
+
+        if (l - rad <= p.x && p.x <= r + rad) {
+          edges.top = (t - rad <= p.y && p.y <= t + rad);
+          edges.bottom = (b - rad <= p.y && p.y <= b + rad);
+        }
+
+        if (t - rad <= p.y && p.y <= b + rad) {
+          edges.left = (l - rad <= p.x && p.x <= l + rad);
+          edges.right = (r - rad <= p.x && p.x <= r + rad);
+        }
+
+        return edges;
+      }
+
+      public _doRender() {
+        super._doRender();
+        if (this.boxVisible()) {
+          var bounds = this.bounds();
+          var t = bounds.topLeft.y;
+          var b = bounds.bottomRight.y;
+          var l = bounds.topLeft.x;
+          var r = bounds.bottomRight.x;
+
+          this._boxEdgeT.attr({
+            x1: l, y1: t, x2: r, y2: t
+          });
+          this._boxEdgeB.attr({
+            x1: l, y1: b, x2: r, y2: b
+          });
+          this._boxEdgeL.attr({
+            x1: l, y1: t, x2: l, y2: b
+          });
+          this._boxEdgeR.attr({
+            x1: r, y1: t, x2: r, y2: b
+          });
+
+          this._boxCornerTL.attr({ cx: l, cy: t });
+          this._boxCornerTR.attr({ cx: r, cy: t });
+          this._boxCornerBL.attr({ cx: l, cy: b });
+          this._boxCornerBR.attr({ cx: r, cy: b });
```

Comment:
I'll be overriding bounds() so that on `XDragBoxLayer`, the y-values are automatically set, and vice-versa for `YDragBoxLayer`.